### ISSUE TITLE
Issue with "docroot/static/style" directory with "theme: foobar"

### ISSUE THREAD

**Issue by troyengel** [nullism/pykwiki#23]
My mission was to take the `default_theme/post.html` and override it so that I could move the Author and Date below the content, not above. As part of this, I started setting up my own `styles/foobar` config (copied from 'classic'). Latest pykwiki from pip (2.1.18) being used.

1. The post.html template contains `{{conf.style_post}}` which turns out to just load the raw content from `styles/foobar/post.html` but doesn't send it through Jinja2, so it gets the raw template inside the parsed template - OK, so the design is this is HTML styling only, no Jinja2 in there.
1. The above led me to reading the code, the `render_theme_template()` routine sends `conf.theme_path` not `conf.style_path` through Jinja2, and because the new default in `config.yaml` is `theme: None` it actually sends the same path (data/default_theme) through it twice, so
1. I set up a `themes/foobar/post.html` and configured `theme: foobar` to align to the code, which worked (pykwiki cache -f) but....
1. ...if `docroot` is missing (my default, it's a Gitlab CI runner so always a clean docker) _and_ `themes/foobar/static/` directory is missing, it causes a python crash right about here when it tries to copy content to static/ and the directory hasn't been made: https://github.com/nullism/pykwiki/blob/master/pykwiki/core.py#L422

So the tl;dr to get where I wanted:

```
make themes/foobar/static/.gitkeep
create themes/foobar/post.html
set theme: foobar
```

The docs for v2 are kind of light, the doc says "theme is now null" but that's about it, so it's hard to tell if this was intentional or not, as it's not really described anywhere (but it works). I think though I found a bug there in the last item, what's actually crashing with a "no such file or directory" is the code right after the copying:

```
        with open(os.path.join(style_dest, 'style.css'), 'w') as fh:
            fh.write(style_data)
```

Because the code above it is only triggered when theme != None, it kind of goes wonky a bit when I set theme = foobar. The logic looks like if it were written to better check `stat_dest` and `style_dest` a little better it might work, but I can see maybe 3 ways to "fix" it depending on design intent.

**Comment by nullism** [comment]
Hi troyengel,

You did the correct thing by creating your own theme. Styles do not (and should not) be parsed by Jinja, as they're just custom HTML to be loaded in the page.

Just to confirm, when you manually added the `static/` directory to your foobar theme, it didn't throw an exception, correct? 

Maybe the correct thing to do is to make a static directory inside the target's docroot when it's not present. I'll look into it a bit more before making the change.

Also I agree that the docs are very light. Shortly after creating Pykwiki2, Gitbook added their search feature, so I assumed that'd become much more popular (Pykwiki's claim to fame was the only static content generator that had built-in search) and didn't spent a lot of time on the docs.

Glad folks are using this; might be time to update the docs.

Thanks for the feedback!

**Comment by nullism** [comment]
I've testing this change and published a new version (2.1.19) to pip. If I understand the issue correctly, this should now be resolved. I fixed a few other issues (trailing whitespace, redeclared variables), but the main change is here: https://github.com/nullism/pykwiki/blob/6ad6f1ec716d9888c49b8851564417154371adaa/pykwiki/core.py#L430

Feel free to re-open if the issue persists and we can work through it.

**Comment by troyengel** [comment]
Working great, thanks!
